lib/transport/websocket.js

```diff
@@ -122,16 +122,27 @@ Factory.prototype.create = function () {
             options.protocol = protocols;
          }
 
-         if (self._options.url.startsWith('wss://')) {
+         if (
+            self._options.url.startsWith('wss://') &&
+            self._options.tlsConfiguration &&
+            self._options.tlsConfiguration.ca &&
```

Comment:
Mmm, do you need these last three lines? I think the error is trying to access an attribute inside a null, i.e. if tlsConfiguration is none, but assuming tlsConfiguration is set, .ca, .auth etc should return null rather than erroring, which should be ok?

Reply:
I don't have a specific reason for this, but if you specify a tlsConfiguration, it should contain all those three items, otherwise it won't work. If not, there is a message in the logs, that TLS client authentication was not used due to invalid options.

Reply:
ok, so I think we can be a little more explicit with messaging there. We should first check if `tlsConfiguration` was provided, if it was, we may then evaluate if all its required parameters were provided as well, in case a parameter is missing we should actually log that, so that it becomes easy for anyone using the library to know why TLS is not being used.